gems/pending/postgres_ha_admin/failover_monitor.rb

```diff
@@ -0,0 +1,122 @@
+require 'postgres_ha_admin/failover_databases'
+require 'postgres_ha_admin/database_yml'
+require 'pg'
+require 'linux_admin'
+
+module PostgresHaAdmin
+  class FailoverMonitor
+    FAILOVER_ATTEMPTS = 60
+    DB_CONNECTED_CHECK_FREQUENCY = 5.minutes
+    FAILOVER_CHECK_FREQUENCY = 1.minute
+
+    def initialize(db_yml_file = '/var.www/miq/vmdb/config/database.yml',
+                   failover_yml_file = '/var.www/miq/vmdb/config/failover_databases.yml',
+                   log_file = '/var.www/miq/vmdb/config/ha_admin.log',
+                   environment = 'production')
+      @logger = Logger.new(log_file)
+      @logger.level = Logger::INFO
+      @database_yml = DatabaseYml.new(failover_yml_file, environment)
+      @failover_db = FailoverDatabases.new(db_yml_file, @logger)
+    end
+
+    def monitor
+      connection = pg_connection(@database_yml.pg_params_from_database_yml)
+      if connection
+        @failover_db.update_failover_yml(connection)
+        connection.finish
+        return
+      end
+
+      @logger.error("Primary Database is not available. EVM server stop initiated. Starting to execute failover...")
+      stop_evmserverd
+
+      if execute_failover
+        start_evmserverd
+      else
+        @logger.error("Failover failed")
+      end
+    end
+
+    def monitor_loop
+      loop do
+        sleep(DB_CONNECTED_CHECK_FREQUENCY)
+        begin
+          monitor
+        rescue StandardError => err
+          @logger.error("#{err.class}: #{err}")
+          @logger.error(err.backtrace.join("\n"))
+        end
+      end
+    end
+
+    def host_for_primary_database(connection, params)
```

Comment:
If we are only using the `host` value from the params hash, I think it would be more clear to only pass that in.

Also it feels like this could be named better and should return true/false because we are not actually getting any new information from the return value.

Maybe `host_is_repmgr_primary?(host, connection)` for the method signature?